src/utils/arweave.ts

```diff
@@ -34,3 +36,16 @@ export const pruneTags = (
     }): tag is { name: string; value: string } => tag.value !== undefined,
   );
 };
+
+export const getCurrentBlockUnixTimestamp = async (
+  arweave: Arweave,
+): Promise<Timestamp> => {
+  return await arweave.blocks
+    .getCurrent()
+    .then((block) => {
+      return block.timestamp * 1000;
+    })
+    .catch(() => {
+      return Date.now(); // fallback to current time
+    });
```

Comment:
nit: I get wary of `return await` by itself, but adding a .then+.catch handler on top of it makes it harder to reason -- and it swallows potential errors the client might want to know. could just be like 
```suggestion
  try {
    const block = await arweave.blocks
    .getCurrent();
    return block.timestamp * 1000
  } catch { // fallback to current time -- consider logging error and fallback intent
    return Date.now();
  }
```

Reply:
these are inconsequential errors - a logger could be passed in to the function.

There is a global `Logger.default` available for use, not too dissimilar from turbos setup.

Reply:
also r.e. try/catch. i understand your point. this is more concise but less readable, fair. but we're not doing anything nuanced here. imo the try/catch pattern is actually HARDER to follow in most scenarios (especially with massive try blocks)